__tests__/__unit__/job/actions.unit.test.ts

```diff
@@ -0,0 +1,109 @@
+import * as vscode from "vscode";
+import * as zowe from "@zowe/cli";
+import { Job } from "../../../src/job/ZoweJobNode";
+import {
+    generateImperativeSession,
+    generateImperativeProfile,
+    generateTreeView
+} from "../../../__mocks__/generators/shared";
+import { generateIJobObject, generateJobsTree } from "../../../__mocks__/generators/jobs";
+import * as jobActions from "../../../src/job/actions";
+
+Object.defineProperty(vscode.window, "showInformationMessage", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn() });
+Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn() });
+Object.defineProperty(zowe, "IssueCommand", { value: jest.fn() });
+Object.defineProperty(zowe.IssueCommand, "issueSimple", { value: jest.fn() });
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = (fn: any): jest.Mock => fn;
+
+describe("Job state operations", () => {
+    function generateEnvironmentalMocks() {
+        const session = generateImperativeSession();
+        const treeView = generateTreeView();
+        const iJob = generateIJobObject();
+        const imperativeProfile = generateImperativeProfile();
+
+        return {
+            session,
+            treeView,
+            iJob,
+            imperativeProfile,
+            testJobsTree: generateJobsTree(session, iJob, imperativeProfile, treeView)
+        };
+    }
+
+    beforeEach(() => {
+        // Reset global module mocks
+        mocked(vscode.window.showInformationMessage).mockReset();
+        mocked(vscode.window.showInputBox).mockReset();
+    });
+
```

Comment:
In each describe block, can we add the following, to make certain that every describe block is self-contained:

```
    afterAll(() => {
        jest.clearAllMocks();
    });
```

Reply:
I was thinking about that, but if we have many mocks declared globally it can be resource-intensive.

Reply:
I still think this is pretty important. This was one of the main issues we were trying to fix with the refactoring, that tests were being affected by previous tests in the same file.

If some test files are running slowly or taking too many resources, we can reexamine how we deal with the mocks in that particular file.

Reply:
Well, if you check my code I add _beforeEach_ and reset only mocks which are used inside of **describe** section. This method though doesn't cover **spyOn**, which is reset in the end of **it** block.